### PR TITLE
examples: go-bench should pass random data

### PR DESCRIPTION
**Pull request by arekkas**
Benchmarks are more impressive with random data!

### CODE REVIEW

File: examples/nats-bench.go [hunk 1]
```diff
@@ -102,14 +103,11 @@ func runPublisher(startwg, donewg *sync.WaitGroup, opts nats.Options, numMsgs in
 
 	args := flag.Args()
 	subj := args[0]
-	var msg []byte
-	if msgSize > 0 {
```
Comment (arekkas) [inline examples/nats-bench.go]: this doesn't make sense as it's the long version of: `var msg = make([]byte, 0)`

**Comment by arekkas** [comment]
Not sure why tests fail on Go 1.6 - works locally.

**Comment by aricart** [comment]
One thing with the random bytes is that while that sounds useful to simulate performance, effectively we are inserting an 'application' delay to generate the bytes, as every message will now have to generate some random number of bytes.

In the bench we are trying to measure the number of bytes/messages that the messaging system can pass through. Additional delays are understood, but not the goal of the bench.

**Comment by arekkas** [comment]
That is very much possible, unfortunately I don't know the internals of the benchmarking tool used.

**Comment by kozlovic** [comment]
I second @aricart comments. Creating a slice in that loop with random data will be really costly and skew the results of the messaging layer that we are benchmarking here. Those slices will put pressure on GC and aren't you going to deplete the entropy pool doing that in a tight loop?

**Comment by arekkas** [comment]
If benchmark includes this loop it doesn't make sense, I thought benchmarking would happen on flush only.

**Comment by kozlovic** [comment]
@arekkas I see. No, the benchmark if measuring all line of codes before the `for` loop (notice the capture of start time). We *could* make ONE message before the capture of the start with a slice of random data, but again, not sure in what this is useful for the purpose of the benchmark.

**Comment by aricart** [comment]
Effectively the publisher is making a byte array of the specified size. So whether random or not, that number of bytes are sent...

**Pull request closed without merging**